### PR TITLE
TestFoundation: TestDateInterval: Add tests for functionality

### PR DESCRIPTION
**Pull request by tillson**
Adds some functionality tests for the DateInterval interface.

* This is my first pull request to this project, so let me know if there are any issues!

### CODE REVIEW

**Comment by tillson** [comment]
@swift-ci please test

**Comment by spevans** [comment]
@swift-ci please test

**Comment by spevans** [comment]
@swift-ci test linux

File: TestFoundation/TestDateInterval.swift [hunk 1]
```diff
@@ -10,10 +10,124 @@
 class TestDateInterval: XCTestCase {
     static var allTests: [(String, (TestDateInterval) -> () throws -> Void)] {
         return [
+            ("test_default_constructor", test_default_constructor),
```
Comment (millenomi) [inline TestFoundation/TestDateInterval.swift]: Can we `camelCase` these?

**Comment by tillson** [comment]
@swift-ci test linux

**Comment by tillson** [comment]
Pardon my Git ignorance, I didn't realize that pushing to the forked master would automatically add 
it to the pull request (I guess the lesson there is to submit a pull request via a branch).

@swift-ci test linux

File: TestFoundation/TestDateInterval.swift [hunk 2]
```diff
@@ -10,10 +10,124 @@
...
+            ("test_intersects", test_intersects),
+            ("test_intersection", test_intersection),
+            ("test_intersectionZeroDuration", test_intersectionZeroDuration),
+            ("test_intersectionNil", test_intersectionNil),
+            ("test_contains", test_contains),
             ("test_hashing", test_hashing),
         ]
     }
 
+    func test_defaultConstructor() {
```
Comment (ikesyo) [inline TestFoundation/TestDateInterval.swift]: We don't use the `constructor` term in Swift. It is called `initializer`: https://docs.swift.org/swift-book/LanguageGuide/Initialization.html

**Comment by spevans** [comment]
@swift-ci test

**Comment by millenomi** [comment]
@swift-ci please test

**Comment by tillson** [comment]
@please test

**Comment by spevans** [comment]
@swift-ci test

**Comment by tillson** [comment]
> @swift-ci test

Is there something wrong?

**Comment by spevans** [comment]
@swift-ci test

**Pull request merged**